django/conf/__init__.py

```diff
@@ -51,7 +51,22 @@ def __repr__(self):
     def __getattr__(self, name):
         if self._wrapped is empty:
             self._setup(name)
-        return getattr(self._wrapped, name)
+        val = getattr(self._wrapped, name)
+        self.__dict__[name] = val  # Cache to avoid getattr call chain again
+        return val
+
+    def __setattr__(self, name, value):
+        if name == '_wrapped':
```

Comment:
Could a docstring to the effect of "Clear the cache if a new _wrapped attribute assigned or if a setting is changed." be an improvement over the `# Clear cache` comment? Similar docstrings for the other methods might help also. As someone who hasn't look at this code before, the high level idea of what's going on seems more difficult to grasp than the implementation details.